### PR TITLE
Branch delete department mergeable

### PR DESCRIPTION
**Pull request by pkwangjoo**
- Support DeleteDepartmentCommand
- Cascade deletion of employee to department

### CODE REVIEW

**Review by kennethk-1201 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/sudohr/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -34,13 +35,19 @@ public DeleteCommand(Index targetIndex) {
...
 
         if (targetIndex.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_EMPLOYEE_DISPLAYED_INDEX);
         }
 
         Employee employeeToDelete = lastShownList.get(targetIndex.getZeroBased());
         model.deleteEmployee(employeeToDelete);
+
+        // cascade delete to department
+        departmentList.stream().forEach(d -> d.removeEmployee(employeeToDelete));
```
[the remark this answers is not in the document]
Comment (pkwangjoo) [reply src/main/java/seedu/sudohr/logic/commands/DeleteCommand.java]: done

File: src/main/java/seedu/sudohr/logic/parser/DeleteDepartmentCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the DeleteDepartmentCommand
+     * and returns a DeleteDepartmentCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public DeleteDepartmentCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_DEPARTMENT_NAME);
+
```
[the remark this answers is not in the document]
Comment (pkwangjoo) [reply src/main/java/seedu/sudohr/logic/parser/DeleteDepartmentCommandParser.java:23]: done

File: src/test/java/seedu/sudohr/logic/commands/DeleteCommandTest.java [hunk 3]
```diff
@@ -76,6 +80,44 @@ public void execute_invalidIndexFilteredList_throwsCommandException() {
...
+                .filter(i -> model.getFilteredEmployeeList().get(i).equals(employeeToDelete))
+                .findFirst()
+                .getAsInt();
+
+        String expectedMessage = String.format(DeleteCommand.MESSAGE_DELETE_EMPLOYEE_SUCCESS, employeeToDelete);
+
+        Model expectedModel = new ModelManager(model.getSudoHr(), new UserPrefs());
+        expectedModel.deleteEmployee(employeeToDelete);
+
+        DeleteCommand deleteCommand = new DeleteCommand(Index.fromZeroBased(indexOfEmployeeToDelete));
```
[the remark this answers is not in the document]
Comment (pkwangjoo) [reply src/test/java/seedu/sudohr/logic/commands/DeleteCommandTest.java:99]: noted

**Review by kennethk-1201 — APPROVED** [review]
LGTM

**Review by 4ndrelim — APPROVED** [review]
LGTM. Indexes changes made on my end.

**Pull request merged**